site/content/blog/carvelizing-helm-chart.md

```diff
@@ -0,0 +1,293 @@
+
+Save the below Package CR to a file named `1.0.0.yaml`
+```yaml
+
+apiVersion: data.packaging.carvel.dev/v1alpha1
+kind: Package
+metadata:
+  # Must be of the form '<spec.refName>.<spec.version>' (Note the period)
+  name: nginx.bitnami.vmware.com.1.0.0
+spec:
+  # The name of the PackageMetadata associated with this version
+  # Must be a valid PackageMetadata name (see PackageMetadata CR for details)
+  # Cannot be empty
+  refName: nginx.bitnami.vmware.com
+  # Package version; Referenced by PackageInstall;
+  # Must be valid semver (required)
+  # Cannot be empty
+  version: 1.0.0
+  # Version release notes (optional; string)
+  releaseNotes:
+    Initial release of the nginx package by wrapping helm Chart. Nginx Helm chart version is 9.5.4
+  # valuesSchema can be used to show template values that
+  # can be configured by users when a Package is installed.
+  # These values should be specified in an OpenAPI schema format. (optional)
+  # For helm chart, we can either define the configurable values here or let it be. Even if we dont define them here, we can still customize them.
+  valuesSchema:
+    openAPIv3:
+      title: nginx.bitnami.vmware.com
+      examples:
+      properties:
+  template:
+    spec:
+      fetch:
+      - helmChart:
+          name: nginx
+          version: 9.5.4
+          repository:
+            # From where to pull the helm chart
+            url: https://charts.bitnami.com/bitnami 
+      template:
+      - helmTemplate: {}
+      deploy:
+      - kapp: {}
+```
+
+**3. Create Package Repository**: A package repository is a collection of packages and their metadata. We will use `imgpkg` bundle to create package repository.
+
+For the purpose of this tutorial, we will run an unsecured local docker registry. In the real world please be safe and use appropriate security measures.
+    
+```bash
+$ docker run -d -p 5000:5000 --restart=always --name registry registry:2
+```
+
+From the terminal we can access this registry as localhost:5000 but within the cluster we'll need the IP Address. We will store the IP address in a variable:
+
+```bash 
+$ export REPO_HOST="`ifconfig | grep inet | grep -E '\b10\.' | awk '{ print $2}'`:5000"
+```
+
+Confirm that REPO_HOST is set to <IP_ADDRESS:PORT>
+
+```bash
+$ echo $REPO_HOST
+  10.104.3.7:5000
+```
+
+Lets start by creating the needed directories:
+
+```bash
+$ mkdir -p nginx-bitnami-repo nginx-bitnami-repo/.imgpkg nginx-bitnami-repo/packages/nginx.bitnami.vmware.com
+```
+
+We can copy our CR YAMLs from the previous step in to the proper packages subdirectory. 
+
+```bash
+$ cp 1.0.0.yaml nginx-bitnami-repo/packages/nginx.bitnami.vmware.com
+$ cp pkgMetadata.yaml nginx-bitnami-repo/packages/nginx.bitnami.vmware.com
+```
+
+Now, let’s use kbld to record which package bundles are used:
+
+```bash
+$ kbld -f nginx-bitnami-repo/packages/ --imgpkg-lock-output nginx-bitnami-repo/.imgpkg/images.yml
+```
+
+We will push the bundle onto repository using `imgpkg`.
```

Comment:
nit: ".... push the bundle into the respository..."

Reply:
Adopted